### PR TITLE
#161342592 A user should be able to login to the system

### PR DESCRIPTION
**Pull request by sharkdevs**
**What does the PR do?**
Provides a login API interface

**Description of the task accomplished**
Creation of the endpoint that allows a given user to login to the system
`POST: /api/v1/users/registration`

**How can this feature be tested**
1. _clone the repository_: create a working folder in a location of choice and in the git bash terminal, type the following command to clone the repository: `$ git clone https://github.com/sharkdevs/store-manager APIs.git`
2. install the virtual environment and activate it using the following commands:
    `$ pip install virtualenv`
    `$ virtualenv venv`
    `$ source venv/Scripts/actvate` for windows
3. install requirements: `pip install -r requirements.txt`
4.  run the application: `$ export FLASK_APP=app` and then `flask run`
5. Run in postman: open postman, copy the server url and append `/api/v1/user/login` to it then past at postman. set the request method as `POST` and run

**What are the relevant pivotaltracker stories**
#161342592

### CODE REVIEW

File: app/api/v1/views.py [hunk 1]
```diff
@@ -0,0 +1,92 @@
+from flask import Flask, make_response, jsonify,request
+from flask_restful import Resource, reqparse
+from flask_jwt_extended import (jwt_required, get_jwt_identity)
+
+
+from app.api.v1.models import Product as p, products
+from app.api.v1.models import Sales as s, sales
```
Comment (Bonifase) [inline app/api/v1/views.py:7]: avoid using single letter variable names like p, s, U, e.t.c. Use more descriptive variable names that make sense.

File: app/tests/v1/test_storemanager.py [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+        #login a sample user
+        self.auth_token = Users.user_login(self, "mesharkz1@gmail.com", "123123")
+
+        #Add a sample product to the list
+        products.append(self.sample_data)
+
+        
+    def test_whether_returns_status_code_on_products_query(self):
+        self.assertEqual(self.app.get('/api/v1/products', headers = dict(Authorization="Bearer " + self.auth_token), content_type='application/json').status_code, 200)
```
Comment (Bonifase) [inline app/tests/v1/test_storemanager.py:39]: Other than testing only status code, also test response messages and data. You can implement this in challenge 3

Comment (Bonifase) [reply app/tests/v1/test_storemanager.py:39]: Looks good. Reduce the number of open PRs at the same time, avoid too many commit messages in a single PR. You can start challenge 3

File: app/api/v1/views.py (hunk 1 above)
Comment (sharkdevs) [reply app/api/v1/views.py:7]: Thank you, am fixing it.

File: app/tests/v1/test_storemanager.py (hunk 2 above)
Comment (sharkdevs) [reply app/tests/v1/test_storemanager.py:39]: Grateful

**Pull request merged**